nixos/modules/security/pam.nix

```diff
@@ -867,34 +944,25 @@ in
 
     environment.etc = mapAttrs' makePAMService config.security.pam.services;
 
-    security.pam.services =
-      { other.text =
-          ''
-            auth     required pam_warn.so
-            auth     required pam_deny.so
-            account  required pam_warn.so
-            account  required pam_deny.so
-            password required pam_warn.so
-            password required pam_deny.so
-            session  required pam_warn.so
-            session  required pam_deny.so
-          '';
-
-        # Most of these should be moved to specific modules.
-        i3lock = {};
-        i3lock-color = {};
-        vlock = {};
-        xlock = {};
-        xscreensaver = {};
-
-        runuser = { rootOK = true; unixAuth = false; setEnvironment = false; };
-
-        /* FIXME: should runuser -l start a systemd session? Currently
-           it complains "Cannot create session: Already running in a
-           session". */
-        runuser-l = { rootOK = true; unixAuth = false; };
-      };
-
+    security.pam.services = {
+      other.entries = let
+        rules = [{ control = "required"; path = "pam_warn.so"; }
+                 { control = "required"; path = "pam_deny.so"; }];
+      in listToAttrs (map (t: nameValuePair t rules) [ "auth" "account" "password" "session" ]);
+
+      # Most of these should be moved to specific modules.
```

Comment:
I wonder where those would end up...

Reply:
I think the original intent was that the kerberos rules would live in their respective packages (e.g. kerberos, LDAP, etc).

Reply:
Indeed.